File: warehouse-fulfilment-java-vt/src/main/java/org/jesperancinha/vtcc/controller/FulfilmentController.java
package org.jesperancinha.vtcc.controller;

import org.jesperancinha.vtcc.domain.Product;
import org.jesperancinha.vtcc.service.ProductService;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;
import java.util.UUID;

@RestController
@RequestMapping("fulfilment")
public class FulfilmentController {

    private final ProductService productService;

    private static final Logger logger = LoggerFactory.getLogger(FulfilmentController.class);

    @Autowired
    public FulfilmentController(ProductService productService) {
        this.productService = productService;
    }

    @GetMapping
    public List<Product> getItems() {
        return productService.getAllItems();
    }

    @GetMapping("{id}")
    public Object getItem(@PathVariable UUID id) {
        return productService.getItemById(id);
    }
}

File: warehouse-fulfilment-vt-coroutines/src/main/kotlin/controller/FulfilmentController.kt
package org.jesperancinha.controller

import org.jesperancinha.domain.Product
import org.jesperancinha.service.ProductService
import org.slf4j.Logger
import org.slf4j.LoggerFactory
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import reactor.core.publisher.Flux
import reactor.core.publisher.Mono
import java.util.UUID

@RestController
@RequestMapping("fulfilment")
class FulfilmentController(
    private val productService: ProductService,
) {

    @GetMapping
    fun getItems(): Flux<Product> = productService.getAllItems()

    @GetMapping("{id}")
     fun getItem(@PathVariable id: UUID): Mono<Product> = productService.getItemById(id)

    companion object {
        val logger: Logger = LoggerFactory.getLogger(FulfilmentController::class.java)
    }
}